poolmgr/gp.go

```diff
@@ -311,12 +344,35 @@ func (gp *GenericPool) specializePod(pod *v1.Pod, metadata *api.ObjectMeta) erro
 
 	// get function run container to specialize
 	log.Printf("[%v] specializing pod", metadata.Name)
-	specializeUrl := gp.getSpecializeUrl(podIP)
 
 	// retry the specialize call a few times in case the env server hasn't come up yet
 	maxRetries := 20
+
+	fn, err = gp.fissionClient.Functions(metadata.Namespace).Get(metadata.Name)
+	if err != nil {
+		return err
+	}
+
+	loadReq := fission.FunctionLoadRequest{
+		FilePath:     filepath.Join(gp.sharedMountPath, "user"),
+		FunctionName: fn.Spec.Package.FunctionName,
+	}
+
+	body, err := json.Marshal(loadReq)
+	if err != nil {
+		return err
+	}
+
 	for i := 0; i < maxRetries; i++ {
-		resp2, err := http.Post(specializeUrl, "text/plain", bytes.NewReader([]byte{}))
+		var resp2 *http.Response
+		// if functionName is empty, assume its environment v1
```

Comment:
Why? I'm not following the logic here...

Reply:
The logic for v2 environments should be simpler: only use the v2 interface if the EnvironmentSpec has Version=2.